src/components/UIShared/ConversationActionsShortcut.vue

```diff
@@ -0,0 +1,152 @@
+import { useRouter, useRoute } from 'vue-router/composables'
+
+import IconCheckUnderline from 'vue-material-design-icons/CheckUnderline.vue'
+import IconDelete from 'vue-material-design-icons/Delete.vue'
+
+import { showError } from '@nextcloud/dialogs'
+import { t, getCanonicalLocale } from '@nextcloud/l10n'
+
+import NcButton from '@nextcloud/vue/components/NcButton'
+import { spawnDialog } from '@nextcloud/vue/functions/dialog'
+
+import ConfirmDialog from '../../components/UIShared/ConfirmDialog.vue'
+
+import { useStore } from '../../composables/useStore.js'
+import { CONVERSATION } from '../../constants.ts'
+import { hasTalkFeature, getTalkConfig } from '../../services/CapabilitiesManager.ts'
+
+const supportsArchive = hasTalkFeature('local', 'archived-conversations-v2')
+const retentionEventPeriod = getTalkConfig('local', 'conversations', 'retention-event')
+const retentionPhonePeriod = getTalkConfig('local', 'conversations', 'retention-phone')
+
+const props = defineProps<{
+	token: string,
+	objectType: string,
+	isHighlighted: boolean,
+}>()
+
+const store = useStore()
+const router = useRouter()
+const route = useRoute()
+
+const isModerator = computed(() => store.getters.isModerator)
+
+const expirationDuration = computed(() => {
+	if (props.objectType === CONVERSATION.OBJECT_TYPE.EVENT) {
+		return retentionEventPeriod
+	} else if (props.objectType === CONVERSATION.OBJECT_TYPE.PHONE_TEMPORARY) {
+		return retentionPhonePeriod
+	}
+	return 0
+})
+
+const descriptionLabel = computed(() => {
+	const expirationDurationFormatted = new Intl.RelativeTimeFormat(getCanonicalLocale(), { numeric: 'auto' }).format(
+		expirationDuration.value, 'days'
+	)
+	return t('spreed', `This conversation will be automatically deleted ${expirationDurationFormatted} of no activity.`)
+})
+
+/**
+ * Delete conversation
+ */
+async function deleteEventConversation() {
+	try {
+		if (route?.params?.token === props.token) {
+			await router.push({ name: 'root' })
+				.catch((failure) => !isNavigationFailure(failure, NavigationFailureType.duplicated) && Promise.reject(failure))
+		}
+		await store.dispatch('deleteConversationFromServer', { token: props.token })
+	} catch (error) {
+		console.error(`Error while deleting conversation ${error}`)
+		showError(t('spreed', 'Error while deleting conversation'))
+	}
+}
+
+/**
+ * Unbind conversation from object
+ */
+async function resetObjectConversation() {
+	await store.dispatch('unbindConversationFromObject', { token: props.token })
+}
+
+/**
+ * Show confirmation dialog
+ */
+async function showConfirmationDialog() {
+	spawnDialog(ConfirmDialog, {
+		name: t('spreed', 'Delete conversation'),
+		message: t('spreed', 'Are you sure you want to delete this conversation?'),
+		buttons: [
+			{
+				label: t('spreed', 'No'),
+				type: 'tertiary',
+			},
+			{
+				label: t('spreed', 'yes'),
+				type: 'error',
+				callback: () => {
+					deleteEventConversation()
+					return true
+				},
+			}
+		],
+	})
```

Comment:
You don't need to return 'true' without onClose callback, only if you code it like this (but there's no need to):
 ```suggestion
					return true
				},
			}
		],
	}, (result) => {
			if (result) {
				deleteEventConversation()
			}
		}
```